src/vs/workbench/contrib/terminal/browser/terminalService.ts

```diff
@@ -507,6 +507,20 @@ export class TerminalService implements ITerminalService {
 		}
 	}
 
+	private async _initAvailableProfiles(): Promise<void> {
+		const result = await this._detectProfiles();
+		const profilesChanged = !equals(result, this._availableProfiles);
+		const contributedProfilesChanged = !equals(this._terminalContributionService.terminalProfiles, this._contributedProfiles);
+		if (profilesChanged || contributedProfilesChanged) {
+			this._availableProfiles = result;
+			this._contributedProfiles = Array.from(this._terminalContributionService.terminalProfiles);
+			this._onDidChangeAvailableProfiles.fire(this._availableProfiles);
+			this._profilesReadyBarrier.open();
+			await this._refreshPlatformConfig(result);
+		}
+	}
```

Comment:
The pattern I usually use for this type of thing is:

```ts
@throttle(2000)
private async _refreshAvailableProfiles(): Promise<void> {
  return this._refreshAvailableProfilesNow();
}

async createTerminal(options?: ICreateTerminalOptions): Promise<ITerminalInstance> {
  if (!this._availableProfiles) {
    await this._refreshAvailableProfilesNow();
  }
  ...
}
```

Reply:
that's what I tried first, but didn't work - see https://github.com/microsoft/vscode/pull/132172#issuecomment-913939637

Reply:
Should work - both `_refreshAvailableProfiles` and `createTerminal` call unthrottled `_refreshAvailableProfilesNow` (notice Now it the end, which is `_initAvailableProfiles` in your code)